packages/react-reconciler/src/ReactFiberApplyGesture.js

```diff
@@ -50,33 +57,295 @@ import {
   OffscreenComponent,
   ViewTransitionComponent,
 } from './ReactWorkTags';
+import {
+  restoreEnterViewTransitions,
+  restoreExitViewTransitions,
+  restoreNestedViewTransitions,
+} from './ReactFiberCommitViewTransitions';
 
 let didWarnForRootClone = false;
 
 function detectMutationOrInsertClones(finishedWork: Fiber): boolean {
   return true;
 }
 
-let unhideHostChildren = false;
+const CLONE_UPDATE = 0; // Mutations in this subtree or potentially affected by layout.
+const CLONE_ENTER = 1; // Inside a reappearing offscreen before the next ViewTransition or HostComponent.
+const CLONE_UNHIDE = 2; // Inside a reappearing offscreen before the next HostComponent.
+const CLONE_APPEARING_PAIR = 3; // Like UNHIDE but we're already inside the first Host Component only finding pairs.
+const CLONE_UNCHANGED = 4; // Nothing in this tree was changed but we're still walking to clone it.
+const INSERT_ENTER = 5; // Inside a newly mounted tree before the next ViewTransition or HostComponent.
+const INSERT_APPEND = 6; // Inside a newly mounted tree before the next HostComponent.
+const INSERT_APPEARING_PAIR = 7; // Inside a newly mounted tree only finding pairs.
+type VisitPhase = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7;
+
+function trackDeletedPairViewTransitions(deletion: Fiber): void {
+  if ((deletion.subtreeFlags & ViewTransitionNamedStatic) === NoFlags) {
+    // This has no named view transitions in its subtree.
+    return;
+  }
+  let child = deletion.child;
+  while (child !== null) {
+    if (child.tag === OffscreenComponent && child.memoizedState === null) {
+      // This tree was already hidden so we skip it.
+    } else {
+      if (
+        child.tag === ViewTransitionComponent &&
+        (child.flags & ViewTransitionNamedStatic) !== NoFlags
+      ) {
+        const props: ViewTransitionProps = child.memoizedProps;
+        const name = props.name;
+        if (name != null && name !== 'auto') {
+          // TODO: Find a pair
+        }
+      }
+      trackDeletedPairViewTransitions(child);
+    }
+    child = child.sibling;
+  }
+}
+
+function trackExitViewTransitions(deletion: Fiber): void {
```

Comment:
Naming becomes interesting for these because they're in reverse. This will actually trigger the `enter` className since deletions are treated as "enter" in reverse. So I'm inclined to rename this to "trackExitViewTransitions" and vice versa.

Reply:
I swapped it.